query-engine/query-engine/src/opt.rs

```diff
@@ -160,22 +160,29 @@ impl PrismaOpt {
     }
 
     /// Extract the log format from on the RUST_LOG_FORMAT env var.
-    pub(crate) fn log_format(&self) -> crate::LogFormat {
+    pub fn log_format(&self) -> crate::LogFormat {
         match self.log_format.as_deref() {
             Some("devel") => crate::LogFormat::Text,
             _ => crate::LogFormat::Json,
         }
     }
 
     /// The unix path to listen on.
-    pub(crate) fn unix_path(&self) -> Option<&String> {
+    pub fn unix_path(&self) -> Option<&String> {
         self.unix_path.as_ref()
     }
 
     /// Enable query logging
-    pub(crate) fn log_queries(&self) -> bool {
+    pub fn log_queries(&self) -> bool {
         std::env::var("LOG_QUERIES").map(|_| true).unwrap_or(self.log_queries)
     }
+
+    // This is added it here to make it easier to create the options when using
+    // the http server in the tests
+    // Ok to unwrap here as this is only used in tests
+    pub fn from_list(list: &[&str]) -> Self {
```

Comment:
This looks like a typical From-trait implementation for me.

Reply:
I can't create a `From` implementation with a fixed size array. So I would have to do that for a vector which seems unnecessary. So I've left it as is.